lightning-invoice/src/payment.rs

```diff
@@ -17,7 +17,7 @@ use lightning::ln::channelmanager::RecipientOnionFields;
 use lightning::routing::router::{PaymentParameters, RouteParameters};
 
 /// Builds the necessary parameters to pay or pre-flight probe the given zero-amount
-/// [`Bolt11Invoice`] using [`ChannelManager::send_payment`] or
+/// (amount-less) [`Bolt11Invoice`] using [`ChannelManager::send_payment`] or
```

Comment:
nit: "amountless" :)

Otherwise, the PR looks good to me. I think you can squash the commits together!

Reply:
Done :)

Reply:
Should I squash refactoring and doc commits together ?

Reply:
Yep!